src/test/java/com/kentyou/featurelauncher/cli/FeatureLauncherCliTest.java

```diff
@@ -0,0 +1,501 @@
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+import static org.osgi.service.featurelauncher.FeatureLauncherConstants.REMOTE_ARTIFACT_REPOSITORY_NAME;
+
+import java.io.ByteArrayOutputStream;
+import java.io.PrintStream;
+import java.net.URISyntaxException;
+import java.nio.file.Paths;
+
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+/**
+ * Tests {@link com.kentyou.featurelauncher.cli.FeatureLauncherCli}
+ * 
+ * As defined in: "160.4.2.4 The Feature Launcher Command Line"
+ * 
+ * @author Michael H. Siemaszko (mhs@into.software)
+ * @since Oct 10, 2024
+ */
+public class FeatureLauncherCliTest {
+	private static final String FEATURE_FILE_ID = "com.kentyou.featurelauncher:gogo-console-feature:1.0";
+	private static final String DECORATOR_CLASS_NAME = "com.kentyou.featurelauncher.impl.decorator.FeatureDecoratorImpl";
+	private static final String EXTENSION_HANDLER_NAME = "LaunchFrameworkFeatureExtensionHandler";
+	private static final String EXTENSION_HANDLER_CLASS_NAME = "com.kentyou.featurelauncher.impl.decorator.LaunchFrameworkFeatureExtensionHandlerImpl";
+	private static final String LAUNCH_PROPERTY_1_KEY = "key1";
+	private static final String LAUNCH_PROPERTY_1_VALUE = "value1";
+	private static final String LAUNCH_PROPERTY_2_KEY = "key2";
+	private static final String LAUNCH_PROPERTY_2_VALUE = "value2";
+	private static final String VARIABLE_1_KEY = "key1";
+	private static final String VARIABLE_1_VALUE = "value1";
+	private static final String VARIABLE_2_KEY = "key2";
+	private static final String VARIABLE_2_VALUE = "value2";
+	private static final String CONFIGURATION_1_KEY = "key1";
+	private static final String CONFIGURATION_1_VALUE = "value1";
+	private static final String CONFIGURATION_2_KEY = "key2";
+	private static final String CONFIGURATION_2_VALUE = "value2";
+
+	private static String FEATURE_FILE_PATH;
+
+	final PrintStream originalOut = System.out;
+	final PrintStream originalErr = System.err;
+	final ByteArrayOutputStream out = new ByteArrayOutputStream();
+	final ByteArrayOutputStream err = new ByteArrayOutputStream();
+
+	@BeforeAll
+	public static void setUpFeatureFilePath() throws URISyntaxException {
+		FEATURE_FILE_PATH = Paths
+				.get(FeatureLauncherCliTest.class.getResource("/features/gogo-console-feature.json").toURI())
+				.toString();
+	}
+
+	@BeforeEach
+	public void setUpStreams() {
+		out.reset();
+		err.reset();
+		System.setOut(new PrintStream(out));
+		System.setErr(new PrintStream(err));
+	}
+
+	@AfterEach
+	public void restoreStreams() {
+		System.setOut(originalOut);
+		System.setErr(originalErr);
+	}
+
+	@Test
+	public void testFeatureFileParameter() {
+		StringBuilder args = new StringBuilder();
+		args.append("--dry-run");
+		args.append(" ");
+		args.append(FEATURE_FILE_PATH);
```

Comment:
This should be the raw feature JSON if it is to be successful, not the path to a file containing the feature JSON